packages/manager/src/features/LoadBalancers/LoadBalancerDetail/ServiceTargets/CreateServiceTargetDrawer.tsx

```diff
@@ -0,0 +1,329 @@
+
+  const {
+    error,
+    mutateAsync: createServiceTarget,
+    reset,
+  } = useServiceTargetCreateMutation(loadbalancerId);
+
+  const formik = useFormik<ServiceTargetPayload>({
+    initialValues,
+    async onSubmit(values) {
+      try {
+        await createServiceTarget(values);
+        onClose();
+      } catch (error) {
+        scrollErrorIntoView();
+      }
+    },
+  });
+
+  const onClose = () => {
+    formik.resetForm();
+    reset();
+    _onClose();
+  };
+
+  const onAddEndpoint = (endpoint: Endpoint) => {
+    formik.setFieldValue('endpoints', [...formik.values.endpoints, endpoint]);
+  };
+
+  const onRemoveEndpoint = (index: number) => {
+    formik.values.endpoints.splice(index, 1);
+    formik.setFieldValue('endpoints', formik.values.endpoints);
+  };
+
+  const generalError = error?.find((e) => !e.field)?.reason;
+
+  return (
+    <Drawer onClose={onClose} open={open} title="Add a Service Target">
+      <form onSubmit={formik.handleSubmit}>
+        {generalError && <Notice text={generalError} variant="error" />}
+        <TextField
+          errorText={error?.find((e) => e.field === 'label')?.reason}
+          label="Service Target Label"
+          name="label"
+          onChange={formik.handleChange}
+          value={formik.values.label}
+        />
+        <Autocomplete
+          errorText={
+            error?.find((e) => e.field === 'load_balancing_policy')?.reason
+          }
+          onChange={(e, selected) =>
+            formik.setFieldValue('load_balancing_policy', selected.value)
+          }
+          renderOption={(props, option, state) => {
+            return (
+              <li {...props}>
+                <Stack flexGrow={1}>
+                  <Typography color="inherit">
+                    <b>{option.label}</b>
+                  </Typography>
+                  <Typography color="inherit">{option.description}</Typography>
+                </Stack>
+                <SelectedIcon visible={state.selected} />
+              </li>
+            );
+          }}
+          value={algorithmOptions.find(
+            (option) => option.value === formik.values.load_balancing_policy
+          )}
+          disableClearable
+          label="Algorithm"
+          options={algorithmOptions}
+        />
+        <Divider spacingBottom={24} spacingTop={24} />
+        <Typography sx={{ marginBottom: 2 }} variant="h3">
+          Endpoints
+        </Typography>
+        <EndpointTable
+          errors={error?.filter((error) =>
+            error.field?.startsWith('endpoints')
+          )}
+          endpoints={formik.values.endpoints}
+          onRemove={onRemoveEndpoint}
+        />
+        <AddEndpointForm onAdd={onAddEndpoint} />
+        <Divider spacingBottom={12} spacingTop={24} />
+        <Stack alignItems="center" direction="row">
+          <Typography variant="h3">Service Target CA Certificate</Typography>
+          <TooltipIcon status="help" text="TODO" />
```

Comment:
```suggestion
          <TooltipIcon status="help" text="TODO: AGLB" />
```
Do we have a follow up ticket in the backlog for this? If so, it would make it easier to search when we come back to it.